src/keychain/index.js

```diff
@@ -503,6 +503,57 @@ class Keychain {
       return throwDelayed(errcode(new Error(`Key '${name}' does not exist. ${err.message}`), 'ERR_KEY_NOT_FOUND'))
     }
   }
+  
+  /**
+   * Rotate keychain password and re-encrypt all assosciated keys
+   *
+   * @param {string} oldPass - The old local keychain password
+   * @param {string} newPass - The new local keychain password
+   */
+  async rotateKeychainPass (oldPass, newPass){
+    if (typeof oldPass !== 'string' || typeof newPass !== 'string') {
+      throw new Error(`Invalid pass type '${typeof oldPass}'`);
+    }
+    if (newPass.length < 20) {
+      throw new Error('pass must be least 20 characters')
+    }
+    this.opts = {
+      dek: Keychain.generateOptions().dek,
+      pass: newPass,
+      datastore: this.store
+    }
+
+    const newDek = newPass
+      ? crypto.pbkdf2(
+        newPass,
+        this.opts.dek.salt,
+        this.opts.dek.iterationCount,
+        this.opts.dek.keyLength,
+        this.opts.dek.hash)
+      : ''
+    privates.set(this, { "dek":newDek })
+
+    var keys = await this.listKeys()
+    await keys.forEach(async key =>{
+      var keyAsPEM = await this._getPrivateKey(key.name)
```

Comment:
this PEM was created with the old dek https://github.com/libp2p/js-libp2p/blob/v0.31.5/src/keychain/index.js#L238. You can look into the export key to see how to get the private key from pem using the previous `dek` and then you need to encrypt it once again as it is done in createKey